com.liferay.blade.upgrade.liferay70/src/com/liferay/blade/upgrade/liferay70/apichanges/MavenPomLegacyDependencies.java

```diff
@@ -0,0 +1,203 @@
+		int corrected = 0;
+		IFile xmlFile = getXmlFile(file);
+		IDOMModel xmlModel = null;
+
+		if (xmlFile != null) {
+			try {
+				xmlModel = (IDOMModel) StructuredModelManager.getModelManager().getModelForEdit(xmlFile);
+
+				List<IDOMElement> elementsToCorrect = new ArrayList<>();
+
+				for (Problem problem : problems) {
+					if (KEY.equals(problem.autoCorrectContext)) {
+						IndexedRegion region = xmlModel.getIndexedRegion(problem.startOffset);
+
+						if (region instanceof IDOMElement) {
+							IDOMElement element = (IDOMElement) region;
+							elementsToCorrect.add(element);
+						}
+					}
+				}
+
+				for (IDOMElement element : elementsToCorrect) {
+					xmlModel.aboutToChangeModel();
+
+					NodeList groudIdList = element.getElementsByTagName("groupId");
+					NodeList artifactIdList = element.getElementsByTagName("artifactId");
+					NodeList versionList = element.getElementsByTagName("version");
+
+					if (groudIdList != null && groudIdList.getLength() == 1 && artifactIdList != null
+							&& artifactIdList.getLength() == 1 && versionList != null && versionList.getLength() == 1) {
+
+						IDOMElement groupIdElement = (IDOMElement) groudIdList.item(0);
+						IDOMElement artifactIdElement = (IDOMElement) artifactIdList.item(0);
+						IDOMElement versionElement = (IDOMElement) versionList.item(0);
+
+						String groupIdContent = groupIdElement.getTextContent();
+						String artifactIdContent = artifactIdElement.getTextContent();
+
+						if (groupIdContent != null && "com.liferay.portal".equals(groupIdContent.trim())
+								&& artifactIdContent != null && artifactIdContent.trim().length() > 0) {
+							String[] result = getFixedArtifactIdAndVersion(artifactIdContent.trim());
+
+							if (result != null) {
+								removeChildren(artifactIdElement);
+								Text artifactIdTextContent = element.getOwnerDocument().createTextNode(result[0]);
+								artifactIdElement.appendChild(artifactIdTextContent);
+
+								removeChildren(versionElement);
+								Text versionTextContent = element.getOwnerDocument().createTextNode(result[1]);
+								versionElement.appendChild(versionTextContent);
+							}
+						}
+					}
+
+					xmlModel.changedModel();
+
+					corrected++;
+				}
+
+				xmlModel.save();
+			} catch (Exception e) {
+				e.printStackTrace();
+			} finally {
+				if (xmlModel != null) {
+					xmlModel.releaseFromEdit();
+				}
+			}
+		}
+
+		if (corrected > 0 && !xmlFile.getLocation().toFile().equals(file)) {
+			try {
+				IO.copy(xmlFile.getContents(), file);
+			} catch (IOException | CoreException e) {
+				throw new AutoMigrateException("Error writing corrected file.", e);
+			}
+		}
+
+		return corrected;
+	}
+
+	private String[] getFixedArtifactIdAndVersion(String artifactId) {
+		if (dependenciesConvertMap == null) {
+			dependenciesConvertMap = new String[5][];
+
+			dependenciesConvertMap[0] = new String[] { "portal-service", "com.liferay.portal.kernel", "2.6.0" };
```

Comment:
the latest service-buidler(1.0.142) build code could match 2.4.0 or higher version of portal-kernel, so we can make it 2.4.0 , if we make it 2.0.0 , it will have compile error after build-service